index.js

```diff
@@ -123,51 +140,176 @@ const getDefaultProps = (path, context) => {
   } else {
     throw new Error(`Could not resolve default Props, ${defaultProps}`);
   }
-};
+}
 
 converters.Program = (path, context) /*: K.Program*/ => {
   let result = {};
   result.kind = 'program';
   result.classes = [];
 
+  // try figure out what the default export is
   path.traverse({
-    ClassDeclaration(path) {
-      if (!isReactComponentClass(path)) return;
-      let classProperties = convertReactComponentClass(path, context);
-
-      result.classes.push(classProperties);
+    ExportDefaultDeclaration(exportPath) {
+      if (exportPath.get('declaration').isIdentifier()) {
+        const isDefaultExport = path =>
+          path.get('id').node &&
+          path.get('id').node.name === exportPath.get('declaration').node.name;
+        path.traverse({
+          'FunctionDeclaration|ArrowFunctionExpression'(functionPath) {
+            if (
+              isDefaultExport(functionPath) &&
+              isReactComponentFunction(functionPath)
+            ) {
+              let functionProperties = convertReactComponentFunction(
+                functionPath,
+                context
+              );
+              result.classes.push(functionProperties);
+            }
+          },
+          VariableDeclaration(variablePath) {
+            const declaration = variablePath.get('declarations.0');
+            if (
+              isDefaultExport(declaration) &&
+              isReactComponentFunction(declaration)
+            ) {
+              let functionProperties = convertReactComponentFunction(
+                declaration.get('init'),
+                context
+              );
+              result.classes.push(functionProperties);
+            }
+          },
+          ClassDeclaration(classPath) {
+            if (
+              isDefaultExport(classPath) &&
+              isReactComponentClass(classPath)
+            ) {
+              let classProperties = convertReactComponentClass(
+                classPath,
+                context
+              );
+              result.classes.push(classProperties);
+            }
+          }
+        });
+      } else {
+        exportPath.traverse({
```

Comment:
Is the logic of this fundamentally the same as the other class traversal method (now line 227)? Can we abstract that out into shared logic?

Similar. I'd love to do some refactoring to make it so there was only one call site for `convertReactComponentClass` and `convertReactComponentFunction`. Possibly build up the array of things to be converted then run on them, rather than converting in place as we find them?

Reply:
Also, the babel handbook notes that if you have nested traversers, you want to declare them in a higher scope so you don't recreate the object every time. The fact we are doing this is hidden in several function layers atm.